### PR TITLE
PP-4280 Reorganise Common classes

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT

- Moved validators to `common.validator`
- Moved exceptions to `common.exception`
- Moved common services to `common.services`
- Moved common models to `common.model` and sub folders
- Moved `service.transaction` to `charge.service`

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/common/exception/ChargeNotFoundRuntimeException.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.pay.connector.exception;
+package uk.gov.pay.connector.common.exception;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/common/exception/ChargeNotFoundRuntimeException.java]:
```suggestion
package uk.gov.pay.connector.charge.exception;
```

File: src/main/java/uk/gov/pay/connector/common/exception/ChargeExpiredRuntimeException.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.pay.connector.exception;
+package uk.gov.pay.connector.common.exception;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/common/exception/ChargeExpiredRuntimeException.java]:
```suggestion
package uk.gov.pay.connector.charge.exception;
```

File: src/main/java/uk/gov/pay/connector/common/exception/GenericGatewayRuntimeException.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.pay.connector.exception;
+package uk.gov.pay.connector.common.exception;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/common/exception/GenericGatewayRuntimeException.java]:
```suggestion
package uk.gov.pay.connector.gateway.exception;
```

File: src/main/java/uk/gov/pay/connector/common/exception/RefundException.java [hunk 4]
```diff
@@ -1,6 +1,6 @@
-package uk.gov.pay.connector.exception;
+package uk.gov.pay.connector.common.exception;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/common/exception/RefundException.java]:
```suggestion
package uk.gov.pay.connector.refund.exception;
```

File: src/main/java/uk/gov/pay/connector/common/model/api/ExternalRefundStatus.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.pay.connector.model.api;
+package uk.gov.pay.connector.common.model.api;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/common/model/api/ExternalRefundStatus.java:1]:
```suggestion
package uk.gov.pay.connector.refund.api;
```

File: src/main/java/uk/gov/pay/connector/common/model/api/ExternalTransactionState.java [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.pay.connector.model.api;
+package uk.gov.pay.connector.common.model.api;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/common/model/api/ExternalTransactionState.java:1]:
```suggestion
package uk.gov.pay.connector.charge.api;
```

**Review by georgeracu — DISMISSED** [review]
Overall a clean PR, only a few places where we can move some of the objects in more specific packages. I used the new Beta functionality of suggesting changes that can be incorporated directly to see how it works.

**Comment by kbottla** [comment]
Thanks @georgeracu. Looks like beta functionality works ok for straightforward and small number of changes. In our case, a commit has been created for each comment addressed and as it is moving to different package, had to fix references and package location in IDE ultimately.

**Review by mrlumbu — APPROVED** [review]

**Pull request merged**